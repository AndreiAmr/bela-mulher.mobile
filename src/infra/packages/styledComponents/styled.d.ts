import 'styled-components/native';

export interface CustomThemeProps {
  borderRadius: string;

  colors: {
    main: string;
    secondary: string;
    gray: string;
    onboarding_background: string;
    purple: string;
  };
}

declare module 'styled-components/native' {
  export interface DefaultTheme extends CustomThemeProps {}
}
