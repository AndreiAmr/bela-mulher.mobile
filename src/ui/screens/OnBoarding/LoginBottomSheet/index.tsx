import React, { forwardRef } from 'react';

import { Modalize } from 'react-native-modalize';
import { Modal, useDisclose } from 'native-base';
import Input from '../../../components/Input';
import Button from '../../../components/Button';
import useLogin from './useLogin';
import * as S from './styles';

const LoginBottomSheet = forwardRef<Modalize>((props, ref) => {
  const {
    items: { email, password, isLoading },
    handlers: { handleChange, handleSubmit },
    errors,
  } = useLogin();
  const { isOpen, onClose } = useDisclose();

  return (
    <Modalize ref={ref} snapPoint={300} avoidKeyboardLikeIOS>
      <S.ContentContainer>
        <Input
          onChange={handleChange('email')}
          value={email}
          label="E-mail"
          placeholder="Digite seu e-mail"
          error={errors.email}
        />
        <Input
          onChange={handleChange('password')}
          value={password}
          label="Senha"
          placeholder="Digite sua senha"
          error={errors.password}
          secureTextEntry
        />
        <S.Divider />
        <Button
          label="Entrar"
          onPress={async () => {
            await handleSubmit();
          }}
          color="purple"
          isLoading={isLoading}
        />
      </S.ContentContainer>

      <Modal isOpen={isOpen} onClose={onClose}>
        <Modal.Content maxWidth="400px">
          <Modal.Header>Autenticado com sucesso!</Modal.Header>
        </Modal.Content>
      </Modal>
    </Modalize>
  );
});

export default LoginBottomSheet;
