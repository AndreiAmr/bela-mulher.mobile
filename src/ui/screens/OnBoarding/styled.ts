import styled from 'styled-components/native';

import { RFValue } from 'react-native-responsive-fontsize';
import { Shadow } from 'react-native-shadow-2';

export const Container = styled.View`
  background-color: ${({ theme }) => theme.colors.onboarding_background};
  flex: 1;

  justify-content: center;
  align-items: center;
`;

export const Title = styled.Text`
  font-size: ${RFValue(32)}px;

  color: ${({ theme }) => theme.colors.main};
  margin-top: 18px;
  font-family: 'Ubuntu-Bold';
  /* border-width: 1px; */

  /* margin-bottom: 40%; */
`;
export const Description = styled.Text`
  font-size: ${RFValue(11)}px;
  /* font-weight: medium; */
  color: ${({ theme }) => theme.colors.gray};
  /* margin-top: 18px; */
  margin-bottom: 40%;
  font-family: 'Ubuntu-Bold';

  font-family: 'Ubuntu-Regular';
`;

export const OptionsContainer = styled.View`
  position: absolute;
  bottom: 0px;
  height: 30%;
  width: 100%;
`;

export const OptionsContent = styled(Shadow)`
  height: 100%;
  width: 100%;
  background: white;
  border-top-left-radius: ${({ theme }) => theme.borderRadius};
  border-top-right-radius: ${({ theme }) => theme.borderRadius};
  padding-horizontal: 39px;
  gap: 20px;

  justify-content: center;
  align-items: center;
`;

export const Image = styled.Image`
  width: 100%;
  height: 30%;
`;
