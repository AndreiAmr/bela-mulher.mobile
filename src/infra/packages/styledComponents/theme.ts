import { CustomThemeProps } from './styled';

export const styledTheme: CustomThemeProps = {
  borderRadius: '20px',
  colors: {
    main: '#ff0',
    secondary: 'bรง',
  },
};
