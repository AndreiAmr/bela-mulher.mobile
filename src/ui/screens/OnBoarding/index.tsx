import React, { useRef } from 'react';
import { Image } from 'react-native';
import { Modalize } from 'react-native-modalize';

import BackgroundImage from '../../assets/images/background-image.jpg';
import OrImage from '../../assets/images/or.png';
import * as S from './styled';
import Button from '../../components/Button';

import LoginBottomSheet from './LoginBottomSheet';

const OnBoarding = () => {
  const modalRef = useRef<Modalize>(null);

  const handleOpenModal = (): void => {
    modalRef.current?.open();
  };

  return (
    <S.Container>
      <S.Image source={BackgroundImage} />
      <S.Title>Espaço Bela Mulher</S.Title>

      <S.Description>
        Gerencie seus agendamentos de forma fácil e prática!
      </S.Description>

      <S.OptionsContainer>
        <S.OptionsContent>
          <Button label="Login" onPress={handleOpenModal} />
          <Image source={OrImage} />
          <Button
            label="Cadastre-se"
            onPress={handleOpenModal}
            color="purple"
          />
        </S.OptionsContent>
      </S.OptionsContainer>

      <LoginBottomSheet ref={modalRef} />
    </S.Container>
  );
};

export default OnBoarding;
