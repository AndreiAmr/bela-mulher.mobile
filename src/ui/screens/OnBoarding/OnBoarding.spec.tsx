import 'react-native';

import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import OnBoarding from '.';
import { ThemeProvider } from 'styled-components/native';
import { styledTheme } from '../../../infra/packages/styledComponents/theme';
import { Provider } from 'react-redux';
import store from '../../../infra/redux/store';

describe('On Boarding Screen', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('Should render correctly', async () => {
    const { toJSON, queryByTestId } = render(
      <Provider store={store}>
        <ThemeProvider theme={styledTheme}>
          <OnBoarding />
        </ThemeProvider>
      </Provider>,
    );

    // const container =
    //   queryByTestId('onboarding-container') ||
    //   getByTestId('onboarding-container');

    const loginButton = queryByTestId('login-button');

    // expect(container).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();

    if (loginButton) {
      fireEvent(loginButton!, 'press');
    }
  });
});
