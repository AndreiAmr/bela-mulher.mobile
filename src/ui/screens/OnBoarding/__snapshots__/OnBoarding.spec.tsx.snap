// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On Boarding Screen Should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
    }
  }
  testID="onboarding-container"
>
  <Image
    source={
      {
        "testUri": "../../../src/ui/assets/images/background-image.jpg",
      }
    }
    style={
      {
        "height": "30%",
        "width": "100%",
      }
    }
  />
  <Text
    style={
      {
        "color": "#00BFFF",
        "fontSize": 22,
        "fontWeight": "bold",
        "marginTop": 18,
      }
    }
  >
    Espaço Bela Mulher
  </Text>
  <Text
    style={
      {
        "color": "#6A6A6A",
        "fontSize": 11,
        "fontWeight": "medium",
        "marginBottom": "40%",
      }
    }
  >
    Gerencie seus agendamentos de forma fácil e prática!
  </Text>
  <View
    style={
      {
        "bottom": 0,
        "height": "30%",
        "position": "absolute",
        "width": "100%",
      }
    }
  />
</View>
`;
